release-notes: added release notes for v4.5.0 (#44)
bisdn/bisdn-linux: release-notes.txt

BISDN Linux
v4.5.0 Release Notes
Date: 2022-03-15

! New in this release
- baseboxd:
  - updated version to 1.11.1
  - added initial support for Broadcom KNET interfaces.
- Linux kernel updated to version 5.15.22

> - iproute2: udpate to version 5.16

Reviewer: i think we also added support for STP on bonded interfaces (https://github.com/bisdn/bisdn.github.io/pull/136)

Reviewer: and we fixed this bug here: https://github.com/bisdn/basebox/issues/364

Author: Mentioned STP support on bonded interfaces, and added the bug fix to list of bugfixes

BISDN Linux
v4.5.0 Release Notes
Date: 2022-03-15

! New in this release
- baseboxd:
  - updated version to 1.11.1

>   - added support for Spanning Tree Protocol (STP, RSTP and MSTP) on bonded interfaces

Reviewer 2: This was already part of 4.4.0 (https://github.com/bisdn/bisdn-linux/blob/v4.4.0/release-notes.txt#L10)

Author: Our testing pipelines however were allowed to fail for bonded STP up until two weeks ago (when we no longer saw failing bonded stp tests), should we reword it to say 'now stable support added'?

Reviewer 2: It feels weird having that tied to the list of baseboxd changes, since this isn't a baseboxd change.

I would maybe add a line to the New in this release along side
```
- stp on bond interfaces is now fully supported and considered stable
```

Do we consider it stable? @jklare

  - added initial support for Broadcom KNET interfaces
- Linux kernel updated to version 5.15.22

> - iproute2: udpated to version 5.16

Reviewer 2: might be worthwhile to add a one liner as ! Changed behaviour that you now need to use `bridge -d vlan` instead of `bridge vlan database` to see per-vlan stp states.

Author: Added section on changed behaviour

! Experimental features

> - OF-DPA: initial support for creating and enabling Broadcom KNET interfaces.

Author: I added both projects

> ! Bug fixes

Reviewer: i think linking to the issues created (and now resolved) on github would be helpful here

Author: Linked to github issues/mr where it made sense

Reviewer: lgtm

- fixed bug related to terminaton MAC entries not being correctly removed on IP deletion. (https://github.com/bisdn/basebox/issues/362)

> - resolved STP issue where the initial STP state was missed, causing ports to forward/flood instead of blocking. (https://github.com/bisdn/bisdn.github.io/pull/136)

Reviewer: i think the link does not fit here

Author: Updated to correct pull request in basebox

> - fixed an issue in where any reachable ip neighbor was assumed to be routable even if a route did not exist.

Reviewer: ```suggestion
- fixed an issue where any reachable ip neighbor was assumed to be routable even if a route did not exist.
```

> - ofdpa: fixed 1G ports on PHY2/3 on platform as4630-54pe.

Reviewer: what was fixed here?

Author: This was the polarity flip fix, the commit message reads:

> The polarity flip needs to be set for the 1g ports on PHY2/PHY3 to work, so set it for all 1G ports.

Reviewer 2: ```suggestion
- ES202 (AS4630-54PE): fixed an issue preventing some of the 1G ports from transmitting packets
```

BISDN Linux
v4.5.0 Release Notes
Date: 2022-03-15

! New in this release

> - baseboxd:
>   - updated version to 1.11.1
>   - added initial support for Broadcom KNET interfaces

Reviewer 2: Since this is now basically duplicated in the experimental features, we might was well drop this and compress this to one line:
```suggestion
- baseboxd: updated to version 1.11.1
```

- Linux kernel updated to version 5.15.22
- iproute2: udpated to version 5.16
- stp on bond interfaces is now fully supported and considered stable

! Experimental features

> - basebox/OF-DPA: initial support added for creating and enabling Broadcom KNET interfaces.

Reviewer 2: ```suggestion
- baseboxd/OF-DPA: initial support added for creating and enabling Broadcom KNET interfaces.
```
